pandas/tests/indexes/test_multi.py

```diff
@@ -327,6 +327,28 @@ def assert_matching(actual, expected):
         assert_matching(ind2.labels, new_labels)
         assert_matching(self.index.labels, labels)
 
+    def test_set_label_distinctly_sized_levels(self):
+        # label changing for levels of different magnitude of categories
+        def assert_equal(actual, expected):
+            assert len(actual) == len(expected)
+            for act, exp in zip(actual, expected):
+                act = np.asarray(act)
+                exp = np.asarray(exp)
+                tm.assert_numpy_array_equal(act, exp, check_dtype=True)
+
+        ind = pd.MultiIndex.from_tuples([(0, i) for i in range(130)])
+        new_labels = range(129, -1, -1)
+        ind_reference = pd.MultiIndex.from_tuples(
+            [(0, i) for i in new_labels])
+
+        # [w/o  mutation]
+        ind2 = ind.set_labels(labels=new_labels, level=1)
+        assert_equal(ind2, ind_reference)
```

Comment:
call these result, then you can simply do

``assert result.equals(expected)``

for the inplace case you need to copy before